Koha/REST/V1/Auth.pm

```diff
@@ -110,6 +116,44 @@ sub authenticate_api_request {
 
     my $spec = $c->match->endpoint->pattern->defaults->{'openapi.op_spec'};
     my $authorization = $spec->{'x-koha-authorization'};
+
+    my $authorization_header = $c->req->headers->authorization;
+    if ($authorization_header and $authorization_header =~ /^Bearer /) {
+        my $server = Net::OAuth2::AuthorizationServer->new;
+        my $grant = $server->client_credentials_grant(Koha::OAuth::config);
+        my ($type, $token) = split / /, $authorization_header;
+        my ($valid_token, $error) = $grant->verify_access_token(
+            access_token => $token,
+        );
+
+        if ($valid_token) {
+            my $patron_id   = Koha::ApiKeys->find( $valid_token->{client_id} )->patron_id;
+            my $patron      = Koha::Patrons->find($patron_id);
+            my $permissions = $authorization->{'permissions'};
+            # Check if the patron is authorized
+            if ( haspermission($patron->userid, $permissions)
```

Comment:
this should really be wrapped in a 'tokenhaspermission' call which limits scope down to the individual access token.

Reply:
I agree, if we implemented the scopes. But they are more complex to implement (we need to create an RFC, cover all resources, write a front-end etc) and in the end, we will still need to check for permissions, right? Because you can grant write permissions on (say) patrons, but we still need to check the patron's permissions, and sysprefs like IndependentBranches.

It can be done. It won't be as simpler as you suggest. And it's feature freeze.